Resolving of '~' in targetPath fails on windows using 8.3 naming format

**Description of the problem**: Resolving of **~** in targetPath leads to problems on windows systems using 8.3 format (see https://docs.microsoft.com/de-de/windows/win32/fileio/naming-a-file?redirectedfrom=MSDN). Current behaviour only works for UNIX-systems and there also the current way of replacing **~** anywhere in the path with System.getProperty("user.home") is not correct. This could be done if the path begins with **~** and is followed by a separator. Otherwise you can't use (valid) file paths containing **~** or the (also valid) UNIX descriptor ~anotheruser/path.

Example of failing resolve of a valid windows path:
_C:\user\abcdef~1\path_
p.e. valid resolve to _C:\user\abcdefghi\path_

**Browser and version**:  Chrome 80 but reproducible with all browsers

**Operating system**: Windows Server 2012 R2 Standard

**WebDriverManager version**: 3.8.1

**WebDriverManager use**: Maven dependency:
```
<dependency>
	<groupId>io.github.bonigarcia</groupId>
	<artifactId>webdrivermanager</artifactId>
	<version>3.8.1</version>
</dependency>
```

**WebDriverManager traces**: N.A.

**Error trace**: 
Caused by: java.lang.IllegalArgumentException: Parameter 'directory' is not a directory
	at org.apache.commons.io.FileUtils.validateListFilesParameters(FileUtils.java:545)
	at org.apache.commons.io.FileUtils.listFiles(FileUtils.java:521)
	at org.apache.commons.io.FileUtils.listFiles(FileUtils.java:691)
	at io.github.bonigarcia.wdm.WebDriverManager.getFilesInCache(WebDriverManager.java:926)
	at io.github.bonigarcia.wdm.WebDriverManager.getDriverFromCache(WebDriverManager.java:869)
	at io.github.bonigarcia.wdm.WebDriverManager.handleCache(WebDriverManager.java:860)
	at io.github.bonigarcia.wdm.WebDriverManager.downloadAndExport(WebDriverManager.java:557)
	at io.github.bonigarcia.wdm.WebDriverManager.manage(WebDriverManager.java:547)
	... 23 more

--- a/src/main/java/io/github/bonigarcia/wdm/Config.java
+++ b/src/main/java/io/github/bonigarcia/wdm/Config.java
@@ -291,10 +291,10 @@ public class Config {
 
         if (resolved != null) {
             path = resolved;
-            if (path.contains(HOME)) {
-                path = path.replace(HOME, System.getProperty("user.home"));
-            }
-            if (path.equals(".")) {
+            // Partial support for Bash tilde expansion: http://www.gnu.org/software/bash/manual/html_node/Tilde-Expansion.html
+            if (path.startsWith(HOME + '/')) {
+                path = Paths.get(System.getProperty("user.home"), path.substring(1)).toString();
+            } else if (path.equals(".")) {
                 path = Paths.get("").toAbsolutePath().toString();
             }
         }

--- a/src/test/java/io/github/bonigarcia/wdm/test/CustomTargetTest.java
+++ b/src/test/java/io/github/bonigarcia/wdm/test/CustomTargetTest.java
@@ -16,6 +16,7 @@
  */
 package io.github.bonigarcia.wdm.test;
 
+import io.github.bonigarcia.wdm.Config;
 import static java.lang.invoke.MethodHandles.lookup;
 import static java.nio.file.Files.createTempDirectory;
 import static org.hamcrest.CoreMatchers.startsWith;
@@ -42,28 +43,55 @@ public class CustomTargetTest {
 
     final Logger log = getLogger(lookup().lookupClass());
 
-    Path tmpFolder;
+    Config globalConfig;
 
     @Before
-    public void setup() throws IOException {
-        tmpFolder = createTempDirectory("").toRealPath();
-        WebDriverManager.globalConfig().setTargetPath(tmpFolder.toString());
-        log.info("Using temporal folder {} as cache", tmpFolder);
+    public void setup() {
+        globalConfig = WebDriverManager.globalConfig();
     }
 
     @Test
-    public void testTargetPath() {
+    public void testTargetPath() throws IOException {
+        Path tmpFolder = createTempDirectory("").toRealPath();
+        globalConfig.setTargetPath(tmpFolder.toString());
+        log.info("Using temporary folder {} as cache", tmpFolder);
         WebDriverManager.chromedriver().setup();
         String binaryPath = WebDriverManager.chromedriver().getBinaryPath();
         log.info("Binary path {}", binaryPath);
         assertThat(binaryPath, startsWith(tmpFolder.toString()));
+        log.info("Deleting temporary folder {}", tmpFolder);
+        WebDriverManager.chromedriver().clearCache();
+    }
+
+    @Test
+    public void testTargetPathContainsTilde() {
+        String customPath = "C:\\user\\abcdef~1\\path";
+        globalConfig.setTargetPath(customPath);
+        String targetPath = globalConfig.getTargetPath();
+        log.info("Using {} got {}", customPath, targetPath);
+        assertThat(targetPath, startsWith(customPath));
+    }
+    
+    @Test
+    public void testTargetPathStartsWithTildeSlash() {
+        String customPath = "~/webdrivers";
+        globalConfig.setTargetPath(customPath);
+        String targetPath = globalConfig.getTargetPath();
+        log.info("Using {} got {}", customPath, targetPath);
+        assertThat(targetPath, startsWith(System.getProperty("user.home")));
+    }
+
+    @Test
+    public void testTargetPathStartsWithTilde() {
+        String customPath = "~webdrivers";
+        globalConfig.setTargetPath(customPath);
+        String targetPath = globalConfig.getTargetPath();
+        log.info("Using {} got {}", customPath, targetPath);
+        assertThat(targetPath, startsWith(customPath));
     }
 
     @After
     public void teardown() throws IOException {
-        log.info("Deleting temporal folder {}", tmpFolder);
-        WebDriverManager.chromedriver().clearCache();
-        WebDriverManager.globalConfig().reset();
+        globalConfig.reset();
     }
-
 }

